bftengine/src/bftengine/InternalReplicaApi.hpp

```diff
@@ -80,10 +80,11 @@ namespace bftEngine
 			virtual void onStatusReportTimer(Time cTime, Timer& timer) = 0;
 			virtual void onSlowPathTimer(Time cTime, Timer& timer) = 0;
 			virtual void onInfoRequestTimer(Time cTime, Timer& timer) = 0;
-			virtual void onDebugStatTimer(Time cTime, Timer& timer) = 0;
 			virtual void onMetricsTimer(Time cTime, Timer& timer) = 0;
 
-
+         #ifdef DEBUG_STATISTICS
+			virtual void onDebugStatTimer(Time cTime, Timer& timer) = 0;
+         #endif /* DEBUG_STATISTICS defined */
```

Comment:
I'm not wild about the idea of this function only being defined if a macro is set at compile time. In my experience, this leads rather quickly to the project only compiling when the flag is set in the default manner. Also, I think it's likely that I'll want to enable/disable statistics output on a deployed cluster, without recompiling. Can we make this a config option instead?

Reply:
As an aside, I made this function dependent on a compile time macro after I found Concord-BFT didn't compile with `DEBUG_STATISTICS` turned off (i.e. not what was previously in the default manner). I can try to make this a config option.

To be completely clear, when you say you want to enable/disable statistics output on a deployed cluster without recompiling, restarting the cluster is still okay, right? I can make this a launch-time parameter, but making it dynamically reconfigurable for a running cluster is not something I feel we have the infrastructure to easily do yet in Concord-BFT.

Reply:
Yes, restarting is fine. I just might not be able to copy a new executable to the environment I'm working with.

Reply:
In my most recent commit, I haveremove the `DEBUG_STATISTICS` compile-time macro and added a `debugStatisticsEnabled` `ReplicaConfig` configuration parameter as you have suggested. Please let me know if this addresses your concerns.

Reply:
Looks great, thank you!